modules/porous_flow/doc/content/modules/porous_flow/thm_example.md

```diff
@@ -0,0 +1,519 @@
+
+!listing modules/porous_flow/examples/thm_example/2D.i start=[./ini_strain] end=[./thermal_contribution]
+
+The seals do not actually have to be considered in the model, because of La Force et al.'s assumptions.
+
+## Boundary conditions
+
+La Force et al. assume the following boundary conditions.
+
+- The porepressure at the outer boundary is
+  \begin{equation}
+    P(r_{\mathrm{max}}) = P_{0} \ .
+  \end{equation}
+
+!listing modules/porous_flow/examples/thm_example/2D.i start=[./outer_pressure_fixed] end=[./outer_temp_fixed]
+
+- The temperature at the outer boundary is
+  \begin{equation}
+    T(r_{\mathrm{max}}) = T_{0} \ .
+  \end{equation}
+
+!listing modules/porous_flow/examples/thm_example/2D.i start=[./outer_temp_fixed] end=[./fixed_outer_r]
+
+- All displacements at the outer boundary are zero
+  \begin{equation}
+    u(r_{\mathrm{max}}) = 0 \ .
+  \end{equation}
+
+!listing modules/porous_flow/examples/thm_example/2D.i start=[./fixed_outer_r] end=[./co2_injection]
+
+- The injection is at a constant rate of $5\times 10^5\,$T.year$^{-1}$ at a constant temperature of 294$\,$K.  In the PorousFlow input file, the fluid injection is implemented as a [PorousFlowSink](boundaries.md), which slowly ramps up to its full value during the initial 100$\,$s in order to help convergence, and a `PresetBC`:
+
+!listing modules/porous_flow/examples/thm_example/2D.i start=[./co2_injection] end=[./cavity_pressure_x]
+
+- The *total* radial compressive stress at the well is equal to the porepressure in the well.  This is dictated by the dynamics of the problem: since the injection rate is constant, the porepressure will increase with time.
+  \begin{equation}
+    \sigma_{rr}^{\mathrm{tot}}(r_{\mathrm{bh}}) = -P \ .
+  \end{equation}
+
+This is implemented in a slightly convoluted way in the MOOSE input file.  First, a `Postprocessor` records the value of porepressure at the beginning of each time step:
+
+!listing modules/porous_flow/examples/thm_example/2D.i block=Postprocessors
+
+Then a [PressureBC](bcs/Pressure.md) applies this *total* stress (physically, a mechanical pushing force) at the borehole wall:
+
+!listing modules/porous_flow/examples/thm_example/2D.i start=[./cavity_pressure_x] end=[]
+
+The `component` is zero because this is an axially-symmetric (2D RZ) problem.  The value of the total pressure should be equal to the variable `pwater`, but because the `p_bh` is recorded at the start of each time step, this boundary condition actually lags by one time step.  Because porepressure `pwater` does not change significantly over each time step, this leads to minimal error.
+
+## Results
+
+!media media/porous_flow/2D_thm_compare_porepressure_fig.png caption=Comparison between the PorousFlow result and the analytic expression derived by La Force et al. for the porepressure id=porepressure_cf_fig
```

Comment:
Maybe add the Eq. number link in the caption (same for other figures) to make it easier for people to get back to the equation?